activiti-services/activiti-services-connectors/src/main/java/org/activiti/services/connectors/channel/ServiceTaskResultHandler.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.services.connectors.channel;
+
+import org.activiti.engine.RuntimeService;
+import org.activiti.services.connectors.model.ServiceTaskResult;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.stereotype.Component;
+
+@Component
+@EnableBinding(ServiceTaskChannels.class)
+public class ServiceTaskResultHandler {
+
+    private final RuntimeService runtimeService;
+
+    @Autowired
+    public ServiceTaskResultHandler(RuntimeService runtimeService) {
+        this.runtimeService = runtimeService;
+    }
+
+    @StreamListener(ServiceTaskChannels.SERVICE_TASK_RESULT_CONSUMER)
+    public synchronized void receive(ServiceTaskResult serviceTaskResult) {
+
+
+        runtimeService.trigger(serviceTaskResult.getExecutionId(),
```

Comment:
@erdemedeiros before calling runtimeService.trigger() we need to do several checks and that's why I think that we need that persistent correlationId that stores the executionId. So the checks should be something like: 
0) a new IntegrationEvent is created with a correlationId, at this stage we need to store that we will be waiting for that correlationId to be completed when an IntegrationResultEvent arrives.
1) a new IntegrationResultEvent arrives we need to check in the DB that the correlationId that the event contains has a matching in the DB
2) load the entry using the correlationId, load the executionId from the DB and then call the runtimeService.trigger() method
3) delete the correlationId entry from the DB because the Service Task (or integration) was completed. 

Does this make sense? 
Again, sending the execution ID out of the Runtime Bundle scope is not a good idea, process id and task id are fine because they are accesible via our public API anyways.

Reply:
It makes sense. I'm going to update that.